src/libraries/System.Security.Cryptography/src/System/Security/Cryptography/X509Certificates/MLDsaX509SignatureGenerator.cs

```diff
@@ -0,0 +1,61 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Diagnostics;
+using System.Formats.Asn1;
+using Internal.Cryptography;
+
+namespace System.Security.Cryptography.X509Certificates
+{
+    internal sealed class MLDsaX509SignatureGenerator : X509SignatureGenerator
+    {
+        private readonly MLDsa _key;
+
+        internal MLDsaX509SignatureGenerator(MLDsa key)
+        {
+            Debug.Assert(key != null);
+
+            _key = key;
+        }
+
+        public override byte[] GetSignatureAlgorithmIdentifier(HashAlgorithmName hashAlgorithm)
+        {
+            // Ignore the hashAlgorithm parameter.
+            // This generator only supports ML-DSA "Pure" signatures, but the overall design of
+            // CertificateRequest makes it easy for a hashAlgorithm value to get here.
+
+            AsnWriter writer = new AsnWriter(AsnEncodingRules.DER);
+            writer.PushSequence();
+            writer.WriteObjectIdentifier(_key.Algorithm.Oid);
+            writer.PopSequence();
+            return writer.Encode();
+        }
+
+        public override byte[] SignData(byte[] data, HashAlgorithmName hashAlgorithm)
+        {
+            ArgumentNullException.ThrowIfNull(data);
+
+            // Ignore the hashAlgorithm parameter.
+            // This generator only supports ML-DSA "Pure" signatures, but the overall design of
```

Comment:
could we check for `is default` here or add HashAlgorithmName.None and throw if not?